compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -672,7 +672,43 @@ pub fn orphan_check(tcx: TyCtxt<'_>, impl_def_id: DefId) -> Result<(), OrphanChe
         return Ok(());
     }
 
-    orphan_check_trait_ref::<!>(trait_ref, InCrate::Local, |ty| Ok(ty)).unwrap()
+    orphan_check_trait_ref::<!>(trait_ref, InCrate::Local, |ty| {
+        let delay_bug = || {
+            tcx.sess.span_delayed_bug(
+                tcx.def_span(impl_def_id),
+                format!(
+                    "orphan check: failed to normalize `{trait_ref}` while checking {impl_def_id:?}"
+                ),
+            )
+        };
+
+        let infcx = tcx.infer_ctxt().intercrate(true).build();
+        let cause = ObligationCause::dummy();
+
+        // FIXME(fmease): This ain't right...
+        let ocx = ObligationCtxt::new(&infcx);
+        let args = infcx.fresh_args_for_item(cause.span, impl_def_id);
+        let ty = ty::EarlyBinder::bind(ty).instantiate(tcx, args);
+        let ty = ocx.normalize(&cause, tcx.param_env(impl_def_id), ty);
```

Comment:
normalize in an empty env, given that we replaced all the params with infer vars